packages/hyperion-autologging/src/ALElementInfo.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+ */
+
+'use strict';
+
+import { ComponentNameValidator, defaultComponentNameValidator, getReactComponentData_THIS_CAN_BREAK } from './ALReactUtils';
+import type { ReactComponentData } from './ALReactUtils';
+import { getVirtualPropertyValue, setVirtualPropertyValue } from '@hyperion/hyperion-core/src/intercept';
+
+const AL_ELEMENT_INFO_PROPNAME = '__alInfo';
+const AUTO_LOGGING_COMPONENT_TYPE = 'data-auto-logging-component-type';
+
+export default class ALElementInfo {
+  element: Element;
+  _reactComponentData: ReactComponentData | null = null;
+  _reactComponentType: string | null = null;
+  _componentNameValidator: ComponentNameValidator = defaultComponentNameValidator;
```

Comment:
Does this need to be stores per instance of this object? I assume this might be per app, and so may be it could be static?

Reply:
Yeah I think that makes sense,  thoughts on if I should create a parent class that underneath manages creating ALElementInfo instances,  where we can keep the object that is set via virtual property as slim as possible?

Reply:
Technically by using the  `static get(..)` method you achieved the same thing. If you wanted to have a parent class, then you had to create a singleton instance of that class to manage things. A (class + singleton) is almost the same as using (class + static) members on it.